### ISSUE TITLE
Support miette::IntoDiagnostic for pest::error::Error<R>

### ISSUE THREAD

**Issue by cgranade** [pest-parser/pest#582]
It would be nice to be able to more easily use [miette](https://docs.rs/miette/) to format diagnostics generated by pest errors. Currently, since `pest::error::Error::message` is private, this cannot easily be done by third-party libraries that depend on both pest and miette without re-implementing much of the message formatting logic in pest. It would be helpful if either `message` was made public, or if there was a feature-gated `impl<R> IntoDiagnostic for pest::error::Error<R>` to make it easier to include pest errors in miette diagnostics.

Thank you! ♥

**Comment by tomtau** [comment]
@bobbbay would you like to tackle this? You mentioned `miette` integration in https://github.com/pest-parser/pest/discussions/606#discussioncomment-3118111

**Comment by bobbbay** [comment]
Yes, I already experimented with this locally. A feature-gated implementation is (in my opinion) the best way to go. The issue is moreso refactoring the internal Error representation so that we can implement miette's `SpanContents` on pest's built-in span type.

I would love to work on this, but I'm currently on vacation and can't always be next to my laptop. If anybody would like to work on this before I get back (end of August), please do.

**Comment by gavrilikhin-d** [comment]
So what’s the status on this?

**Comment by bobbbay** [comment]
The PR is closed and I don't plan on working on this anytime soon. From an implementation standpoint, it's doable.
